src/components/TablePagination/TablePagination.test.tsx

```diff
@@ -0,0 +1,84 @@
+/* eslint-disable testing-library/no-node-access */
+import React from "react";
+import { render, screen, fireEvent } from "@testing-library/react";
+import TablePagination from "./TablePagination";
+import userEvent from "@testing-library/user-event";
+
+const dummyData = [
+  { id: "row-1" },
+  { id: "row-2" },
+  { id: "row-3" },
+  { id: "row-4" },
+  { id: "row-5" },
+];
+
+describe("<TablePagination />", () => {
+  // snapshot tests
+  it("renders table pagination and matches the snapshot", () => {
+    render(<TablePagination data={dummyData} />);
+
+    expect(screen.getByRole("navigation")).toMatchSnapshot();
+  });
+
+  // unit tests
+  it("renders default display title correctly when no pagination takes place", () => {
+    render(<TablePagination data={dummyData} />);
+
+    expect(screen.getByRole("navigation")).toHaveTextContent(
+      "Showing all 5 items"
+    );
+  });
+
+  it("renders default display title correctly when pagination takes place", () => {
+    render(<TablePagination data={dummyData} pageLimits={[1]} />);
+
+    expect(screen.getByRole("navigation")).toHaveTextContent(
+      "Showing 1 out of 5 items"
+    );
+  });
+
+  it("has correct per page setting when changed", () => {
+    render(<TablePagination data={dummyData} pageLimits={[2, 5, 10]} />);
+
+    expect(screen.getByRole("navigation")).toHaveTextContent("2/page");
+    userEvent.selectOptions(
+      screen.getByRole("combobox", { name: "Items per page" }),
+      "5"
+    );
+    expect(screen.getByRole("navigation")).toHaveTextContent("5/page");
+  });
+
+  it("resets to first page when page size is changed", () => {
+    render(<TablePagination data={dummyData} pageLimits={[2, 5, 10]} />);
+
+    expect(screen.getByRole("navigation")).toHaveTextContent("2/page");
+    userEvent.selectOptions(
+      screen.getByRole("combobox", { name: "Items per page" }),
+      "5"
+    );
+    const currentPageInput = screen.getByRole("spinbutton", {
+      name: "Page number",
+    });
+    expect(currentPageInput).toHaveValue(1);
```

Comment:
First switch to 2nd page, because 1 is the default page already.

Reply:
I actually wanted to check that it starts off on the 1st page because it is the default, then decrementing the page number should result in nothing happening since you can't decrement to less than 1. Do you think maybe the first check for page === 1 is redundant?

Reply:
First check is fine, I was expecting this to be more extensive: go to 2nd (or 3rd) page, then change page size and then ensure the page is set back to 1. Currently, the test would pass if the component had a bug and wouldn't reset back to 1 on page size change, because the test doesn't ever leave the 1st page.

Reply:
sure, will add more detailed steps to the test as you mentioned :+1:

Reply:
I have added some more details to this test

Reply:
It was added to the other test case, but I guess its fine this way too.

Reply:
Oh my bad, I didn't look at the line number here and thought you were talking about a different test. I guess the case is covered nonetheless